tfgrid3/mastodon/Dockerfile

```diff
@@ -1,42 +1,59 @@
-    wget -q http://www.webmin.com/jcameron-key.asc -O- | sudo apt-key add - &&\
-    add-apt-repository "deb [arch=amd64] http://download.webmin.com/download/repository sarge contrib" &&\
-    apt install -y webmin
-
-RUN wget https://github.com/cdr/code-server/releases/download/v3.3.1/code-server-3.3.1-linux-amd64.tar.gz &&\
+    $(lsb_release -cs) stable" | tee /etc/apt/sources.list.d/docker.list > /dev/null && \
+    # Docker-compose
+    curl -L "https://github.com/docker/compose/releases/download/1.29.2/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose && \
+    chmod +x /usr/local/bin/docker-compose && \
+    # Zinit
+    wget -O /sbin/zinit https://github.com/threefoldtech/zinit/releases/download/v0.2.5/zinit && \
+    chmod +x /sbin/zinit && \
+    # Webmin
+    wget -q http://www.webmin.com/jcameron-key.asc -O- | apt-key add - && \
+    add-apt-repository "deb [arch=amd64] http://download.webmin.com/download/repository sarge contrib" && \
+    # Codeserver
+    wget https://github.com/cdr/code-server/releases/download/v3.3.1/code-server-3.3.1-linux-amd64.tar.gz &&\
     tar -xzvf code-server-3.3.1-linux-amd64.tar.gz &&\
-    sudo cp -r code-server-3.3.1-linux-amd64 /usr/lib/code-server &&\
-    sudo ln -s /usr/lib/code-server/bin/code-server /usr/bin/code-server &&\
-    sudo mkdir /var/lib/code-server
-
-RUN sudo apt install -y debian-keyring debian-archive-keyring &&\
-    curl -1sLf 'https://dl.cloudsmith.io/public/caddy/stable/gpg.key' | sudo tee /etc/apt/trusted.gpg.d/caddy-stable.asc &&\
-    curl -1sLf 'https://dl.cloudsmith.io/public/caddy/stable/debian.deb.txt' | sudo tee /etc/apt/sources.list.d/caddy-stable.list &&\
-    curl https://dl.cloudsmith.io/public/caddy/stable/gpg.key | sudo gpg --yes --dearmor --output /usr/share/keyrings/caddy-stable-archive-keyring.gpg &&\
-    echo 'deb [signed-by=/usr/share/keyrings/caddy-stable-archive-keyring.gpg] https://dl.cloudsmith.io/public/caddy/stable/deb/debian any-version main' | sudo tee /etc/apt/sources.list.d/caddy-stable.list &&\
-    sudo apt update &&\
-    sudo apt install caddy
+    cp -r code-server-3.3.1-linux-amd64 /usr/lib/code-server &&\
+    ln -s /usr/lib/code-server/bin/code-server /usr/bin/code-server &&\
+    mkdir /var/lib/code-server && \
+    # Cady 
+    curl -1sLf 'https://dl.cloudsmith.io/public/caddy/stable/gpg.key' | tee /etc/apt/trusted.gpg.d/caddy-stable.asc &&\
+    curl -1sLf 'https://dl.cloudsmith.io/public/caddy/stable/debian.deb.txt' | tee /etc/apt/sources.list.d/caddy-stable.list &&\
+    curl https://dl.cloudsmith.io/public/caddy/stable/gpg.key | gpg --yes --dearmor --output /usr/share/keyrings/caddy-stable-archive-keyring.gpg &&\
+    echo 'deb [signed-by=/usr/share/keyrings/caddy-stable-archive-keyring.gpg] https://dl.cloudsmith.io/public/caddy/stable/deb/debian any-version main' | tee /etc/apt/sources.list.d/caddy-stable.list && \
+    # Update and install all required packages.
+    apt-get update --fix-missing && \
```

Comment:
why is this `--fix-missing` option useful here?

Reply:
removed